pymc_marketing/mmm/fourier.py

```diff
@@ -0,0 +1,653 @@
+
+        """
+        full_period = np.arange(self.days_in_period + 1)
+        coords = {
+            "day": full_period,
+        }
+        for key, values in parameters[self.variable_name].coords.items():
+            if key in {"chain", "draw", self.prefix}:
+                continue
+            coords[key] = values.to_numpy()
+
+        with pm.Model(coords=coords):
+            name = f"{self.prefix}_trend"
+            pm.Deterministic(
+                name,
+                self.apply(dayofyear=full_period),
+                dims=tuple(coords.keys()),
+            )
+
+            return pm.sample_posterior_predictive(
+                parameters,
+                var_names=[name],
+            ).posterior_predictive[name]
+
+    def plot_full_period(
+        self,
+        curve: xr.DataArray,
+        subplot_kwargs: dict | None = None,
+        sample_kwargs: dict | None = None,
+        hdi_kwargs: dict | None = None,
+    ) -> tuple[plt.Figure, npt.NDArray[plt.Axes]]:
+        """Plot the seasonality for one full period.
+
+        Parameters
+        ----------
+        curve : xr.DataArray
+            Sampled full period of the fourier seasonality.
+        subplot_kwargs : dict, optional
+            Keyword arguments for the subplot, by default None
+        sample_kwargs : dict, optional
+            Keyword arguments for the plot_full_period_samples method, by default None
+        hdi_kwargs : dict, optional
+            Keyword arguments for the plot_full_period_hdi method, by default None
+
+        Returns
+        -------
+        tuple[plt.Figure, npt.NDArray[plt.Axes]]
+            Matplotlib figure and axes.
+
+        """
+        hdi_kwargs = hdi_kwargs or {}
+        sample_kwargs = sample_kwargs or {}
+
+        if "subplot_kwargs" not in hdi_kwargs:
+            hdi_kwargs["subplot_kwargs"] = subplot_kwargs
+
+        fig, axes = self.plot_full_period_hdi(curve, **hdi_kwargs)
+        fig, axes = self.plot_full_period_samples(curve, axes=axes, **sample_kwargs)
+
+        return fig, axes
+
+    def plot_full_period_hdi(
+        self,
+        samples: xr.DataArray,
+        hdi_kwargs: dict | None = None,
+        axes: npt.NDArray[plt.Axes] | None = None,
+        subplot_kwargs: dict[str, Any] | None = None,
+        plot_kwargs: dict[str, Any] | None = None,
+    ) -> tuple[plt.Figure, npt.NDArray[plt.Axes]]:
+        """Plot full period of the fourier seasonality.
+
+        Parameters
+        ----------
+        parameters : az.InferenceData | xr.Dataset
+            Inference data or dataset containing the fourier parameters.
+            Can be posterior or prior.
+        hdi_prob : float, optional
+            HDI probability, by default 0.95
+        ax : plt.Axes, optional
+            Matplotlib axes, by default None
+        **kwargs
+            Additional keyword arguments for `fill_between`.
+
+        Returns
+        -------
+        plt.Axes
+            Modified matplotlib axes.
```

Comment:
This does not match the functions signature

Reply:
good catch. Was older version
Switched up